trie/database.go

```diff
@@ -782,7 +785,7 @@ func (c *cleaner) Put(key []byte, rlp []byte) error {
 	delete(c.db.dirties, hash)
 	c.db.dirtiesSize -= common.StorageSize(common.HashLength + int(node.size))
 	if node.children != nil {
-		c.db.dirtiesSize -= common.StorageSize(cachedNodeChildrenSize + len(node.children)*(common.HashLength+2))
+		c.db.childrenSize -= common.StorageSize(cachedNodeChildrenSize + len(node.children)*(common.HashLength+2))
```

Comment:
Why do we need to change this?

Reply:
This was an issue I recognized while looking into MPT fullnode and it was fixed in upstream after l2geth was forked. It is not a requirement, but not a big change either.